pkg/cloudprovider/cloud.go

```diff
@@ -0,0 +1,104 @@
+package cloudprovider
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/golang/glog"
+)
+
+// Interface defines functions that need to be implemented by cloudproviders.
+type Interface interface {
```

Comment:
I think it's too early to have a cloud provider interface. Best we could probably do is create boundaries between the Kubernetes client code and the cloud provider code.

Can we build something first, then try to figure out what interfaces we need? Interfaces usually make more sense when the fall out naturally from the use cases rather than trying to define them up front.